src/gui/guiFormSpecMenu.cpp

```diff
@@ -766,108 +766,75 @@ void GUIFormSpecMenu::parseScrollBarOptions(parserData* data, const std::string
 
 void GUIFormSpecMenu::parseImage(parserData* data, const std::string &element)
 {
-	std::vector<std::string> parts = split(element,';');
-
-	if ((parts.size() == 3) ||
-		((parts.size() > 3) && (m_formspec_version > FORMSPEC_API_VERSION)))
-	{
-		std::vector<std::string> v_pos = split(parts[0],',');
-		std::vector<std::string> v_geom = split(parts[1],',');
-		std::string name = unescape_string(parts[2]);
+	std::vector<std::string> parts = split(element, ';');
 
-		MY_CHECKPOS("image", 0);
-		MY_CHECKGEOM("image", 1);
+	if (parts.size() < 3 || parts.size() > 4 ||
+			(parts.size() > 4 && m_formspec_version <= FORMSPEC_API_VERSION)) {
+		errorstream << "Invalid image element(" << parts.size() << "): '" << element
+			<< "'" << std::endl;
+		return;
+	}
 
-		v2s32 pos;
-		v2s32 geom;
+	std::vector<std::string> v_pos = split(parts[0],',');
+	std::vector<std::string> v_geom = split(parts[1],',');
```

Comment:
Doesn't this break compat for images without a size?

Reply:
It's undocumented, and I've never seen it in any mods, so I removed it.